src/NuGet.Core/NuGet.Build.Tasks/GenerateTemporaryTargetAssembly2.cs

```diff
@@ -0,0 +1,523 @@
+        /// ITask Execute method
+        /// </summary>
+        /// <returns></returns>
+        public override bool Execute()
+        {
+            bool retValue = true;
+
+            // Verification
+            try
+            {
+                XmlDocument xmlProjectDoc = null;
+
+                xmlProjectDoc = new XmlDocument();
+                xmlProjectDoc.Load(CurrentProject);
+
+                //
+                // remove all the WinFX specific item lists
+                // ApplicationDefinition, Page, MarkupResource and Resource
+                //
+
+                RemoveItemsByName(xmlProjectDoc, APPDEFNAME);
+                RemoveItemsByName(xmlProjectDoc, PAGENAME);
+                RemoveItemsByName(xmlProjectDoc, MARKUPRESOURCENAME);
+                RemoveItemsByName(xmlProjectDoc, RESOURCENAME);
+
+                // Replace the Reference Item list with ReferencePath
+
+                RemoveItemsByName(xmlProjectDoc, REFERENCETYPENAME);
+                AddNewItems(xmlProjectDoc, ReferencePathTypeName, ReferencePath);
+
+                // Add GeneratedCodeFiles to Compile item list.
+                AddNewItems(xmlProjectDoc, CompileTypeName, GeneratedCodeFiles);
+
+
+                // Create a random file name
+                // This can fix the problem of project cache in VS.NET environment.
+                //
+
+                // GetRandomFileName( ) could return any possible file name and extension, but
+                // some file externsion has special meaning in MSBUILD system, such as a ".sln"
+                // means the file is a solution file with special file format. Since the temporary
+                // file is just for a project, we can use a fixed extension here, but the basic 
+                // file name is still random which can fix above VS.NET bug.
+                //
+                string randProjPath = Path.ChangeExtension(Path.GetRandomFileName(), Path.GetExtension(CurrentProject));
+
+                // Save the xmlDocument content into the temporary project file.
+                xmlProjectDoc.Save(randProjPath);
+
+                //
+                // Invoke MSBUILD engine to build this temporary project file.
+                //
+
+                Hashtable globalProperties = new Hashtable(3);
+
+                // Add AssemblyName and IntermediateOutputPath to the global property list
+                globalProperties[intermediateOutputPathPropertyName] = IntermediateOutputPath;
+                globalProperties[assemblyNamePropertyName] = AssemblyName;
+                globalProperties[originalProjectNamePropertyName] = Path.GetFileNameWithoutExtension(CurrentProject);
+
+                retValue = BuildEngine.BuildProjectFile(randProjPath, new string[] { CompileTargetName }, globalProperties, null);
+
+                try
+                {
+                    // Delete the random project file from disk.
+                    File.Delete(randProjPath);
```

Comment:
wish list: based on an environment variable, skip the step of deleting the random project. *so many times* I've wished I could inspect and build this file myself when debugging why the inner build fails when the outer build would succeed.

Reply:
Adding code to see if an environment variable of the name "WPF_DoNotDeleteTemporaryProject" has a non-empty/non-null value.